src/libstd/dynamic_lib.rs

```diff
@@ -263,6 +266,16 @@ mod dl {
                  symbol: *const libc::c_char) -> *mut libc::c_void;
         fn dlclose(handle: *mut libc::c_void) -> libc::c_int;
     }
+
+    #[cfg(target_os = "openbsd")]
```

Comment:
Judging by the man page on OpenBSD you don't need to explicitly link with `-ldl`: http://www.openbsd.org/cgi-bin/man.cgi/OpenBSD-current/man3/dladdr.3?query=dlopen&sec=3